ca/certificate-authority_test.go

```diff
@@ -411,6 +415,22 @@ func TestCapitalizedLetters(t *testing.T) {
 	test.AssertDeepEquals(t, expected, parsedCert.DNSNames)
 }
 
+func TestWrongSignature(t *testing.T) {
+	ctx := setup(t)
+	defer ctx.cleanUp()
+	ctx.caConfig.MaxNames = 3
+	ca, err := NewCertificateAuthorityImpl(ctx.caConfig, ctx.fc, ctx.stats, caCertFile)
+	ca.Publisher = &mocks.Publisher{}
+	ca.PA = ctx.pa
+	ca.SA = ctx.sa
+
+	// x509.ParseCertificateRequest() does not check for invalid signatures...
+	csr, _ := x509.ParseCertificateRequest(WrongSignatureCSR)
```

Comment:
Sorry, doesn't throw a panic. It flips the `failed` bit and exits the current goroutine with `runtime.Goexit`.